File: python/high-scores/high_scores.py
from heapq import nlargest
from typing import List

Scores = List[int]


def latest(scores: Scores) -> int:
    """The last added score."""
    return scores[-1]


def personal_best(scores: Scores) -> int:
    """The highest score."""
    return max(scores)


def personal_top_three(scores: Scores) -> Scores:
    """The three highest scores."""
    return nlargest(3, scores)

File: ruby/high-scores/high_scores.rb
# Manage a game player's High Score list
class HighScores
  attr_reader :scores, :latest, :personal_best, :personal_top_three

  def initialize(scores)
    @scores = scores
    @latest = scores.last
    @personal_best = scores.max
    @personal_top_three = scores.max(3)
  end
end
